### PR TITLE
Fix for some projects with unusual node_modules structure

### PR DESCRIPTION
**Pull request by antonina-cherednichenko**
There are some projects with folder node_modules which does not have submodules with external libs, but only javascript files inside it. Package.json file can be also absent.

### CODE REVIEW

**Comment by alexsaveliev** [comment]
> There are some projects with folder node_modules which does not have submodules with external libs, but only javascript files inside it. Package.json file can be also absent.

Name them, please

File: lib/analysis.js [hunk 1]
```diff
@@ -186,10 +186,28 @@ function getExternalRepoInfo(data) {
...
+
     var pathRes = suffix.split("/");
-    var packageJsonPath = path.join(prefix,  "node_modules", pathRes[0], "package.json");
+    var dirPath = path.join(prefix,  "node_modules", pathRes[0]);
+
+    var packageJsonPath = "";
+    if (fs.lstatSync(dirPath).isDirectory()) {
+        packageJsonPath = path.join(prefix, "node_modules", pathRes[0], "package.json");
+    } else {
+        packageJsonPath = path.join(prefix, "node_modules", "package.json");
```
Comment (alexsaveliev) [inline lib/analysis.js]: `./node_modules/package.json`? Really?

File: lib/analysis.js [hunk 2]
```diff
@@ -186,10 +186,28 @@ function getExternalRepoInfo(data) {
...
     }
+
     var prefix = filePath.substring(0, pos);
     var suffix = filePath.substring(pos + "node_modules/".length);
+
     var pathRes = suffix.split("/");
-    var packageJsonPath = path.join(prefix,  "node_modules", pathRes[0], "package.json");
+    var dirPath = path.join(prefix,  "node_modules", pathRes[0]);
+
+    var packageJsonPath = "";
```
Comment (alexsaveliev) [inline lib/analysis.js]: useless assignment

File: lib/analysis.js [hunk 3]
```diff
@@ -186,10 +186,28 @@ function getExternalRepoInfo(data) {
...
+    } else {
+        packageJsonPath = path.join(prefix, "node_modules", "package.json");
+    }
+
+
+    //check whether package.json file exists in the determined directory
+    try {
+        fs.statSync(packageJsonPath);
+    } catch (e) {
+        return null;
```
Comment (alexsaveliev) [inline lib/analysis.js]: Does `fs.statSync` throw error?

File: lib/analysis.js [hunk 4]
```diff
@@ -186,10 +186,28 @@ function getExternalRepoInfo(data) {
...
+
     var prefix = filePath.substring(0, pos);
     var suffix = filePath.substring(pos + "node_modules/".length);
+
     var pathRes = suffix.split("/");
-    var packageJsonPath = path.join(prefix,  "node_modules", pathRes[0], "package.json");
+    var dirPath = path.join(prefix,  "node_modules", pathRes[0]);
+
+    var packageJsonPath = "";
+    if (fs.lstatSync(dirPath).isDirectory()) {
```
Comment (alexsaveliev) [inline lib/analysis.js]: Please add comments describing this `if`

**Comment by antonina-cherednichenko** [comment]
Reproduced on 'webpack', 'node-browserify', ....

**Pull request merged**